core/modules/layout/layout.admin.inc

```diff
@@ -584,9 +584,14 @@ function layout_settings_form_update_layout($form, &$form_state) {
     $layout->name = $form_state['values']['name'];
   }
   if (isset($form_state['values']['layout_template'])) {
-    $new = $form_state['values']['layout_template'];
-    $region_mapping = layout_get_legacy_region_mapping($layout->layout_template, $new);
-    $layout->setLayoutTemplate($form_state['values']['layout_template'], $region_mapping);
+    $old_template = $layout->layout_template;
+    $new_template = $form_state['values']['layout_template'];
+    if ($old_template != $new_template) {
+      $region_mapping = layout_get_legacy_region_mapping($old_template, $new_template);
+      $layout->setLayoutTemplate($form_state['values']['layout_template'], $region_mapping);
```

Comment:
why not `$layout->setLayoutTemplate($new_template, $region_mapping)`?

Reply:
Indeed, why not. Fixed1 😉